src/Illuminate/Database/Eloquent/Factories/Factory.php

```diff
@@ -330,7 +330,7 @@ public function lazy(array $attributes = [], ?Model $parent = null)
      */
     protected function store(Collection $results)
     {
-        $results->each(function ($model) {
+        $results->each(function ($model, int $index) {
```

Comment:
```suggestion
        $results->each(function ($model, $index) {
```

We don't normally add strict type on closure.